docs/HowToFAQ/Errors-Resolver3.md

```diff
@@ -0,0 +1,678 @@
+  var i: int;
+  assert fresh(i);
+}
+```
+
+The result of the `fresh` predicate is true if the argument has been allocated since the pre-state of the 
+two-state context containing the call. Thus the argument must be of a type that is allocatable,
+such as a class type --- but not value types like `bool` or `int` or datatypes. The argument may also be
+a set or sequence of such allocatable objects.
+
+## **Error: logical/bitwise negation expects a boolean or bitvector argument (instead got _type_)**
+
+```dafny
+const x := !1
+```
+
+The logical negation operator (`!`) applies to `bool` and bitvector values, but not to, for example,
+`int` values without an explicit conversion. In particular there is no conversion between 0 and false
+or 1 and true.
+
+## **Error: size operator expects a collection argument (instead got _type_)**
+
+```dafny
+method m(x: array<int>) {
+  var y: int := |x|;
+}
+```
+
+The |...| operator is the _size_ operator and returns the integer that is the size of its argument.
+Only finite collections (of type `seq`, `set`, `multiset`, `map`) may be the argument of the
+size operator -- not arrays, `iset`, or `imap`.
+
+## **Error: a _what_ definition is not allowed to depend on the set of allocated references**
+
+<!-- TODO -->
+
+## **Error: type conversion to an int-based type is allowed only from numeric and bitvector types, char, and ORDINAL (got _type_)**
+
+```dafny
+const x: int := true as int
+```
+
+Not all pairs of types have implicit or even explicit conversions. But there are conversions
+to int types from numeric types, including the ORDINAL type; for any source type, the value of 
+the numeric expression must be in the range for the int type (if it is a subset type or a newtype).
+Even `char` values have an integer representation (and thus a represetnation as a `real`) 
+corresponding to their unicode value.
+
+## **Error: type conversion to a real-based type is allowed only from numeric and bitvector types, char, and ORDINAL (got _type_)**
+
+```dafny
+const x: real := true as real
+```
+
+Not all pairs of types have implicit or even explicit conversions. But there are conversions
+to real types from numeric types, including the ORDINAL type; for any source type, the value of 
+the numeric expression must be in the range for the int type (if it is a subset type or a newtype).
+Even `char` values have an integer representation corresponding to their unicode value.
+
+
+## **Error: type conversion to a bitvector-based type is allowed only from numeric and bitvector types, char, and ORDINAL (got _type_)**
+
+```dafny
+const x: bv1 := true as bv1
+```
+
+Not all pairs of types have implicit or even explicit conversions. But there are explicit conversions
+to bitvector types from numeric types, including the ORDINAL type; for any source type, the value of 
+the numeric expression must be in the range for the bitvector type. Even `char` values have an
+integer representation corresponding to their unicode value.
```

Comment:
```suggestion
bitvector representation corresponding to their unicode value.
```
With "a" instead of "an" on the previous line.

Reply:
Fixed